qabelbox/src/main/java/de/qabel/qabelbox/services/LocalQabelService.java

```diff
@@ -144,8 +167,17 @@ public Contacts getContacts() {
      * This should only be used in testing.
      */
     public void deleteContactsAndIdentities() {
-        persistence.dropTable(Contacts.class);
-        persistence.dropTable(Identity.class);
+        try {
+            for (Identity identity: identityRepository.findAll().getIdentities()) {
+                Set<Contact> contacts = contactRepository.find(identity).getContacts();
+                for (Contact contact: contacts) {
+                    contactRepository.delete(contact, identity);
+                }
+                identityRepository.delete(identity);
+            }
+        } catch (EntityNotFoundExcepion | PersistenceException e) {
+            Log.e(TAG, "Error deleting contacts and identities");
```

Comment:
why don't you rethrow it (or change the signature)? especially if it is only used in testing, knowing that something failed (by failing the test) is important. Other behavior will be harder to debug